google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java

```diff
@@ -373,6 +373,11 @@ public static GoogleCredentialsProvider.Builder defaultCredentialsProviderBuilde
         .setJwtEnabledScopes(JWT_ENABLED_SCOPES);
   }
 
+  @Override
+  public String getServiceName() {
+    return "bigtable";
+  }
```

Comment:
I noticed that EnhancedBigtableStubSettings inherits directly from StubSettings and not BigtableStubSettings: https://github.com/googleapis/java-bigtable/blob/26d5437903d5a611f079bb659ca5924227f0aae9/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java#L95

EnhancedBigtableStubSettings will not be able to determine the serviceName because that is generated inside BigtableStubSettings. I have duplicated this here the EnhancedStubSettings can determine the serviceName.

Reply:
IIRC, StubSettings is a public abstract class that is not marked with `@InternalApi`. We were afraid that there may be some use external use cases where users have extended or implemented StubSettings and now would need to implement `getServiceName()`. We provided a default implementation so that it would not break anyone.